runtime/ms_rest/lib/ms_rest/serialization.rb

```diff
@@ -391,7 +391,14 @@ def serialize_sequence_type(mapper, object, object_name)
       # @param model_name [String] Name of the model to retrieve.
       #
       def get_model(model_name)
-        Object.const_get(@context.class.to_s.split('::')[0...-1].join('::') + "::Models::#{model_name}")
+        consts = @context.class.to_s.split('::')
+        if consts.any?{ |const| const == 'Models' }
```

Comment:
The way this logic has been updated, it seems like there was a bug in discovering the model class. Just curious as it seems like you ran into it. :)